test/Adapter/PhpSerializeTest.php

```diff
@@ -122,15 +122,19 @@ public function testUnserializeObject()
         $this->assertEquals($expected, $data);
     }
 
-    public function testUnserializingNonserializedStringReturnsItVerbatim()
+    public function testUnserializingNoStringRaisesException()
     {
-        $value = 'not a serialized string';
-        $this->assertEquals($value, $this->adapter->unserialize($value));
+        $value = null;
+        $this->setExpectedException(
+            'Zend\Serializer\Exception\RuntimeException',
+            'Serialized data must be a string'
+        );
+        $this->adapter->unserialize($value);
     }
 
     public function testUnserializingInvalidStringRaisesException()
     {
-        $value = 'a:foobar';
+        $value = 'foobar';
```

Comment:
I'd argue the change in this test is indicative of exactly the situation I outlined above. If anything, this test should be expanded to use a data provider to try several different strings.